pkg/noderesourcetopology/cache/overreserve.go

```diff
@@ -0,0 +1,234 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	k8scache "k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+
+	topologyv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+	listerv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/listers/topology/v1alpha1"
+
+	"sigs.k8s.io/scheduler-plugins/pkg/noderesourcetopology/stringify"
+
+	"github.com/k8stopologyawareschedwg/podfingerprint"
+)
+
+type OverReserve struct {
+	lock             sync.Mutex
+	nrts             *nrtStore
+	assumedResources map[string]*resourceStore // nodeName -> resourceStore
+	// nodeDiscarded counts how many times a node is filtered out. This is used as trigger condition to try
+	// to resync nodes. See The documentation of Resync() below for more details.
+	nodeDiscarded *counter
+	nrtLister     listerv1alpha1.NodeResourceTopologyLister
+	nodeIndexer   NodeIndexer
+}
+
+func NewOverReserve(lister listerv1alpha1.NodeResourceTopologyLister, indexer NodeIndexer) (Interface, func(), error) {
```

Comment:
nit: given `NewOverReserve` is just a concrete function, it's fine to return `*OverReserve` instead of `Interface`, so that we can shorten the return args as `*OverReserve, error` (the sync func can be get from `*OverReserve`).

Reply:
fixed with the very last upload!